web3/_utils/request.py

```diff
@@ -106,30 +127,78 @@ def make_post_request(
 
 # --- async --- #
 
-_async_session_cache_lock = threading.Lock()
+
 _async_session_cache = SessionCache(size=20)
+_async_session_cache_lock = threading.Lock()
+_pool = ThreadPoolExecutor(max_workers=1)
 
 
 async def cache_async_session(endpoint_uri: URI, session: ClientSession) -> None:
-    cache_key = generate_cache_key(endpoint_uri)
-    with _async_session_cache_lock:
-        evicted_items = _async_session_cache.cache(cache_key, session)
-        if evicted_items is not None:
-            for key, session in evicted_items.items():
-                await session.close()
+    await get_async_session(endpoint_uri, session)
```

Comment:
Since this is just calling through to `get_async_session`, does it make sense to combine `cache_async_session` with `get_async_session`?

Reply:
I think it does... I hadn't touched that part of the code base yet but if @dbfreem is good with it, I lean towards that as well. That abstraction doesn't seem warranted.

Reply:
Hmm since this is merged into `v5` already and exposed... I don't mind the distinction of `get_async_session()` (which actually returns the session) vs `cache_async_session()` which effectively does the same thing without returning. I'm not opposed to changing this in the `v6` PR though. Thoughts there?

Reply:
Ahh, that makes sense. Yeah, let's change here and we'll keep it in v5.